pkg/3scale/amp/component/backend_hpa.go

```diff
@@ -0,0 +1,119 @@
+package component
+
+import (
+	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	hpa "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (backend *Backend) BackendListenerHpa(namespace string, minPods *int32, maxPods int32, cpuPercent *int32, memoryPercent *int32) *hpa.HorizontalPodAutoscaler {
```

Comment:
IMO it would be better to have a single function for creating HPAs with addition of one extra field - Name.
No need to be replicating the function 3 times just to change the name IMO.

Reply:
Dup will address comment in the first one